lib/cn/node_split.c

```diff
@@ -219,18 +254,55 @@ static void
 forward_wbt_leaf_iterator_init(
     struct forward_wbt_leaf_iterator *const iter,
     struct kvset *const ks,
-    const uint32_t kblk_idx)
+    const uint32_t kblk_idx,
+    const void *inflection_key,
+    const uint16_t inflection_key_len)
 {
+    struct key_obj inflection_kobj;
+    struct kvset_kblk *kblk;
+    struct wbt_desc *desc;
+
     INVARIANT(ks);
     INVARIANT(iter);
     INVARIANT(kblk_idx < ks->ks_st.kst_kblks);
 
     iter->ks = ks;
-    iter->eof = false;
-    /* Using prefix decrement in next, so this is fine */
     iter->offset.kblk_idx = kblk_idx;
     iter->offset.leaf_idx = 0;
     iter->es = es_make(forward_wbt_leaf_iterator_next, NULL, NULL);
+
+    if (iter->offset.kblk_idx >= ks->ks_st.kst_kblks)
+        return;
+
+    kblk = &iter->ks->ks_kblks[iter->offset.kblk_idx];
+    desc = &kblk->kb_wbt_desc;
+
+    inflection_kobj.ko_pfx = inflection_key;
+    inflection_kobj.ko_pfx_len = ks->ks_pfx_len;
```

Comment:
Can we do key2kobj() and not rely on ks_pfx_len? I presume we are holding on to this prefix.length cparam purely for ptombs.. May not be a good idea to add another dependency here...

Reply:
Yes, I don't know that was a thing, so I'll change it up.